fiche: init at version 0.9.1

pinpox opened the pull request:
###### Motivation for this change
Include [fiche](https://github.com/solusipse/fiche), a command line pastebin service for sharing terminal output with minimal dependencies.

**Note**: this is my first PR to nixpkgs, feedback is welcome so please let me know what has to be changed!

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/servers/fiche/default.nix:
```diff
@@ -0,0 +1,36 @@
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    description = "Command line pastebin for sharing terminal output.";
```
```suggestion
    description = "Command line pastebin for sharing terminal output";
```

SuperSandro2000 commented on pkgs/servers/fiche/default.nix:
```diff
@@ -0,0 +1,36 @@
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mv fiche $out/bin
```
You can use install here to also set the permission correct.

pinpox replied:
Could you elaborate? I don't understand what I have to change.

turboMaCk replied:
see https://github.com/solusipse/fiche/blob/master/Makefile#L9

pinpox replied:
So just replace those two lines with this?
```
install -m 0755 fiche $out
```

pinpox replied:
I have pushed a commit with the suggested changes

SuperSandro2000 commented on pkgs/servers/fiche/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+  src = fetchurl {
+    url = "https://github.com/solusipse/${pname}/archive/${version}.tar.gz";
+    sha256 = "TODO";
```
If you try to built this locally you will get the Sha that needs to be put here.

pinpox replied:
Done.

SuperSandro2000 commented on pkgs/servers/fiche/default.nix:
```diff
@@ -0,0 +1,36 @@
+  pname = "fiche";
+  version = "0.9.1";
+
+  src = fetchurl {
```
Please use ``fetchFromGitHub``.

pinpox replied:
Updated.

pinpox commented:
The package builds and works fine now on my machine.

pinpox commented:
I'd like to add a service aswell, so that this can be used declaratively from your `configuration.nix` like:
```nix
services = {
  fiche = {
    enable = true;
    domain = "example.com";
    listen_addr  = "0.0.0.0";
    port = "9999";
    slug size = 3;
    output directory = "/var/dir/";
    buffer size = 2048;
    user name = fiche;
    log file = home/www/fiche-log.txt;
    banlist = [];
    whitelist = [];
    https = true; # -S command line option
  };
};
```

Should that be a separate PR? and how/where do I add such a configuration option?

turboMaCk commented:
I would recommend doing separate PR for service. It will likely require other set of reviewers. I would also recommend trying to first define service as part of your personal configuration first and only then trying to port it to nixos modules.

turboMaCk commented on pkgs/servers/fiche/default.nix:
```diff
@@ -0,0 +1,37 @@
+  };
+
+  installPhase = ''
+    install -m 0755 fiche $out/bin
```
```suggestion
    mkdir -p $out/bin
    install -m 0755 fiche $out/bin
```

to avoid:

```
install: cannot create regular file '/nix/store/z026bmn69l1lbjgv09vgikqq67b8bnwr-fiche-0.9.1/bin': No such file or directory
```

turboMaCk commented:
@pinpox Please squash commits into a single one with the message `fiche: init at version 0.9.1` and I think this is good to go from my point of view.

pinpox committed:
fiche: init at version 0.9.1

pinpox commented:
> @pinpox Please squash commits into a single one with the message `fiche: init at version 0.9.1` and I think this is good to go from my point of view.

I found a small bug, fixed now and squashed.

turboMaCk commented:
yup that's cool. I also found it but looks I didin't have to point it out https://github.com/NixOS/nixpkgs/pull/101469#discussion_r510972076

:+1: good work on this. It will probably take a few days for committers to merge this because they're quite overloaded with PRs so please be patient.

sikmir commented on pkgs/servers/fiche/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m 0755 fiche $out/bin
```
```suggestion
    install -m 0755 fiche -t $out/bin
```

pinpox committed:
Update pkgs/servers/fiche/default.nix

Co-authored-by: Nikolay Korotkiy <sikmir@gmail.com>

pinpox committed:
Update default.nix

pinpox commented:
Can someone merge this or comment on what is missing?

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -6751,6 +6751,16 @@
       fingerprint = "A3A3 65AE 16ED A7A0 C29C  88F1 9712 452E 8BE3 372E";
     }];
   };
+  pinpox = {
```
Please put this into a separate commit with the message ``maintainers: add pinpox``.

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -142,6 +142,8 @@ in
 
   ankisyncd = callPackage ../servers/ankisyncd { };
 
+  fiche = callPackage ../servers/fiche { };
```
It would be great if you could sort this into a fitting category and alphabetically there.

SuperSandro2000 reviewed (changes requested):
Also please squash the commits related to the package together into the init commit.

otherwise LGTM

SuperSandro2000 commented:
Result of `nixpkgs-review pr 101469` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiche</li>
  </ul>
</details>

SuperSandro2000 commented:
Result of `nixpkgs-review pr 101469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiche</li>
  </ul>
</details>
